src/convertToObject.js

```diff
@@ -7,6 +7,21 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  const rules = {};
+
+  sourceString
+    .trim()
+    .split(';')
+    .forEach((rule) => {
+      const [key, value] = rule
+        .trim()
+        .split(':')
+        .map((part) => part.trim());
```

Comment:
[CODE KNOWLEDGE] - There's no need to make a copy of the array before operations like `trim` on each part. The `map` method does not mutate the original array, so you can safely use it without copying.